### DISCUSSION TITLE
How to remove specific characters in the copy button?

### DISCUSSION THREAD

**Discussion by pgierz** [executablebooks/meta#756] [Q&A] [answered]
Hello,

Is there a way to remove the starting prompts (or other symbols) in the copy-button? This works in `conf.py` [as follows I believe](https://sphinx-copybutton.readthedocs.io/en/latest/use.html#strip-and-configure-input-prompts-for-code-cells):

```python
copybutton_prompt_text = r">>> |\.\.\. |\$ |In \[\d*\]: | {2,5}\.\.\.: | {5,8}: "
copybutton_prompt_is_regexp = True
```

However, I am unsure how to include this in `_conf.yaml` (is that the right place?)

Any help would be greatly appreciated!
Thanks
Paul

**Comment by choldgraf** [comment, marked as answer]
I think that you could achieve this with [manually configuring Sphinx via Jupyter Book](https://jupyterbook.org/en/stable/advanced/sphinx.html#manual-sphinx-configuration). Does that work?

**Reply by pgierz** [reply]
Yes! That works at least when the `conf.py` is generated; and then readthedocs uses that one. Here are the changes I did to my `_conf.yaml`, for reference:

```diff
sphinx:
  extra_extensions:
    - sphinxcontrib.youtube
+  recursive_update: true
+  config:
+    copybutton_prompt_text: 'r">>> |\.\.\. |\$ |In \[\d*\]: | {2,5}\.\.\.: | {5,8}: "'
+    copybutton_prompt_is_regexp: True
```

@choldgraf: maybe that can go into the documentation somewhere? I am happy to contribute if someone let's me know the best place for that.

**Reply by choldgraf** [reply]
what do you believe was the missing piece? We are trying not to put anything Sphinx-specific in the Jupyter Book documentation, _except for the "Advanced Sphinx usage"_ section. The hope is that if somebody knows what they want in `conf.py`, they can figure out how to add it by coming across that documentation, rather than having Sphinx-specific examples scattered throughout. Do you have a suggestion for the improvement to make in order for this to be more discoverable?

**Reply by pgierz** [reply]
Having examples scattered around everywhere would be confusing, I completely agree. Jupyterbook seems to however have a few things enabled by default (e.g. tabs, copy-button, and so on). Maybe an idea could be to include that in a section for each of those plugins? For example, in https://sphinx-copybutton.readthedocs.io/en/latest/index.html, a section "Using with Jupyter-book: this is enabled by default in Jupyter book. The customizations described in ...some link... can be set in the Jupyter book _conf.yaml as follows blahblahblah" 

I understand though that it is mixing up where best to put the info.

**Reply by choldgraf** [reply]
I think we are trying to move to a model that "compartmentalizes" the documentation in Jupyter Book to more cleanly map onto individual extensions, so even if they're not explicitly mentioning Sphinx, at least we can keep the documentation structure very similar and easier to maintain, I think that would help here too
